src/patterns/strategy/interfaces.ts

```diff
@@ -0,0 +1,17 @@
+export interface IBaseDiscountStrategy {
+  name: string;
+  calculateDiscount(price: number): number;
+}
+
+export interface IFixedDiscountStrategy extends IBaseDiscountStrategy {
```

Comment:
I do not like keeping all the interfaces in the same module, cause the strategy abstraction should not be aware of diffrerent implementations.